### PR TITLE
First draft of chart service. Fix #1131

### LINKED ISSUE #1131: Adding chart support from snap-physics

### PR DESCRIPTION
**Pull request by ellienguyen**

### CODE REVIEW

**Comment by brollb** [comment]
Depends on #1164

File: src/server/rpc/procedures/chart/chart.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+const ApiConsumer = require('../utils/api-consumer'),
+    ChartNode = require('chartjs-node');
+
+let chart = new ApiConsumer('chart');
```
Comment (brollb) [inline src/server/rpc/procedures/chart/chart.js]: why are you subclassing `ApiConsumer`? It doesn't look like you are using any external api....

File: src/server/rpc/procedures/chart/chart.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+let chartNode = new ChartNode(600, 600);
+let defaultColor = 'rgba(74, 108, 212, 0.8)';
+
+let defaultOption = (xAxis, yAxis, title) => {
+    return {
+        scales: {
+            xAxes: [{
```
Comment (brollb) [inline src/server/rpc/procedures/chart/chart.js]: xAxis

File: src/server/rpc/procedures/chart/chart.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                scaleLabel: {
+                    display: true,
+                    labelString: xAxis,
+                    fontSize: 16
+                }
+            }],
+            yAxes: [{
```
Comment (brollb) [inline src/server/rpc/procedures/chart/chart.js]: yAxis

File: package.json [hunk 4]
```diff
@@ -33,6 +33,8 @@
     "cache-manager": "^2.0.0",
     "cache-manager-fs": "^1.0.6",
     "chalk": "^1.1.1",
+    "chart.js": "^2.5.0",
```
Comment (brollb) [inline package.json]: are you using this dependency?

Comment (ellienguyen) [inline package.json]: The chartjs node still need chart.js

File: src/server/rpc/procedures/chart/chart.js (hunk 2 above)
Comment (ellienguyen) [inline src/server/rpc/procedures/chart/chart.js]: That's in the Chart.js documentation
image

**Comment by ellienguyen** [comment]
Here is the request for chart @brollb

File: src/server/rpc/procedures/chart/chart.js (hunk 2 above)
Comment (brollb) [inline src/server/rpc/procedures/chart/chart.js]: ok, nevermind then :)

File: package.json (hunk 4 above)
Comment (brollb) [inline package.json]: weird

**Comment by brollb** [comment]
Could you add some tests for this?

**Comment by ellienguyen** [comment]
The example for this service will be shared with ThingSpeak service examples #1126

**Comment by hamidzr** [comment]
a simple test project.
[chartTest.zip](https://github.com/NetsBlox/NetsBlox/files/1125595/chartTest.zip)
could you please take care of these:
- [ ] when passing multiple more than one dataset the service throws an error
- [ ] number of datasets can be inferred from the structured that is passed in, so it probably shouldn't need user input for that
thanks

**Comment by hamidzr** [comment]
a charting service can be more robust by having features like: 
- checking the xAxis labels to see what type they are, number ? time ? label? and then maybe sort. 
- check for duplicate labels 
but I think we should have this merged in and improve later.

**Comment by brollb** [comment]
deepinscreenshot_select-area_20170727112922

**Comment by brollb** [comment]
looks like I was missing `chart.js`

**Comment by brollb** [comment]
moved to https://github.com/NetsBlox/NetsBlox/pull/1352

**Pull request closed without merging**